lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -2089,53 +2094,95 @@ diagnoseMatch(ModuleDecl *module, NormalProtocolConformance *conformance,
+    auto witness = match.Witness;
+    auto diag = diags.diagnose(
+        witness, diag::protocol_witness_prefix_postfix_conflict, true,
+        witness->getAttrs().hasAttribute<PrefixAttr>() ? 1 : 0);
+    if (witness->getAttrs().hasAttribute<PrefixAttr>()) {
+      auto attr = witness->getAttrs().getAttribute<PrefixAttr>();
+      diag.fixItReplace(attr->getLocation(), "postfix");
+    }
 
-  case MatchKind::PostfixNonPostfixConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness,
-                   diag::protocol_witness_prefix_postfix_conflict, true,
-                   match.Witness->getAttrs().hasAttribute<PrefixAttr>() ? 1
-                                                                        : 0);
     break;
-  case MatchKind::MutatingConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness,
-                   diag::protocol_witness_mutation_modifier_conflict,
-                   SelfAccessKind::Mutating);
+  }
+  case MatchKind::MutatingConflict: {
+    auto witness = match.Witness;
+    auto diag = diags.diagnose(
+        witness, diag::protocol_witness_mutation_modifier_conflict,
+        SelfAccessKind::Mutating);
+    auto attr = witness->getAttrs().getAttribute<MutatingAttr>();
+    diag.fixItRemove(attr->getLocation());
+
     break;
-  case MatchKind::NonMutatingConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness,
-                   diag::protocol_witness_mutation_modifier_conflict,
-                   SelfAccessKind::NonMutating);
+  }
+  case MatchKind::NonMutatingConflict: {
+    auto witness = match.Witness;
+    auto diag = diags.diagnose(
+        witness, diag::protocol_witness_mutation_modifier_conflict,
+        SelfAccessKind::NonMutating);
+    auto attr = witness->getAttrs().getAttribute<NonMutatingAttr>();
+    diag.fixItRemove(attr->getLocation());
     break;
-  case MatchKind::ConsumingConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness,
-                   diag::protocol_witness_mutation_modifier_conflict,
-                   SelfAccessKind::Consuming);
+  }
+  case MatchKind::ConsumingConflict: {
+    auto witness = match.Witness;
+    auto diag = diags.diagnose(
+        witness, diag::protocol_witness_mutation_modifier_conflict,
+        SelfAccessKind::Consuming);
+    auto attr = witness->getAttrs().getAttribute<ConsumingAttr>();
+    diag.fixItRemove(attr->getLocation());
     break;
-  case MatchKind::RethrowsConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness, diag::protocol_witness_rethrows_conflict);
+  }
+  case MatchKind::RethrowsConflict: {
+    auto witness = match.Witness;
+    auto diag =
+        diags.diagnose(match.Witness, diag::protocol_witness_rethrows_conflict);
+    assert(isa<FuncDecl>(witness));
+    auto FD = cast<FuncDecl>(witness);
+    diag.fixItReplace(FD->getThrowsLoc(), getTokenText(tok::kw_rethrows));
```

Comment:
This does seem a little more probable: someone doesn't know about `rethrows` yet but they happen to have implemented their function that way.